### ISSUE TITLE
gitsm (in kirkstone) supports all git options incl. user

### ISSUE THREAD

**Issue by jo-so** [priv-kweihmann/oelint-adv#784]
At least in Kirkstone, the code of gitsm:// looks like it supports all options of git:// including the *user* option. But oelint-adv complains:
```
% oelint-adv test.bb 
test.bb:1:warning:oelint.vars.srcurioptions:Option 'user' is not known with this fetcher type [branch:true]
% cat test.bb
SRC_URI = "gitsm://git.example.org/foo.git;protocol=ssh;user=git"
% oelint-adv --version
oelint-adv 8.2.0
```

From poky/bitbake/lib/bb/fetch2/gitsm.py:

```python
"""
BitBake 'Fetch' git submodules implementation
 
Inherits from and extends the Git fetcher to retrieve submodules of a git repository
after cloning.
 
SRC_URI = "gitsm://<see Git fetcher for syntax>"
 
See the Git fetcher, git://, for usage documentation.
"""

class GitSM(Git):
    def download(self, ud, d):
… 
        Git.download(self, ud, d)
```

**Comment by priv-kweihmann** [comment]
Instead of filling up issues, I'd appreciate if that is really an issue, patches would be contributed.


This project is basically a one man show, so any help is appreciated
